apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -150,111 +38,82 @@ suite('Contacts', function(done) {
     window.fb = mockReaderUtils;
   });
 
-  teardown(function() {
-    navigator.mozContacts.mHistory.length = 0;
-  });
-
   suiteTeardown(function() {
     navigator.mozContacts = nativeMozContacts;
     window.fb = realFb;
+    //Restoring the native API
+    navigator.mozContacts.find.restore();
   });
 
   suite('Contacts.findContactByString, single-term', function() {
 
     test('(string[tel,givenName,familyName], ...) Match', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('O\'Hare').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
```

Comment:
last proposal, if the filter needs to be more complex sometimes:

function findWithFilter(filter) {
  return navigator.mozContacts.find.withArgs(
    sinon.match(filter)
  );
}

// in the test
findWithFilter({filterValue: 'O\'Hare'}).returns(Promise.resolve(result));
```